frontend/viewer/src/lib/writing-system-service.ts

```diff
@@ -0,0 +1,135 @@
+﻿import type {IEntry, IExampleSentence, IMultiString, ISense, IWritingSystem, IWritingSystems} from '$lib/dotnet-types';
+import {getContext, setContext} from 'svelte';
+import {derived, get, type Readable} from 'svelte/store';
+import type {WritingSystemSelection} from './config-types';
+import {firstTruthy} from './utils';
+
+const writingSystemContextName = 'writingSystems';
+export function initWritingSystemService(ws: Readable<IWritingSystems | null>): Readable<WritingSystemService | null> {
+  return setContext(writingSystemContextName, derived(ws, ($ws) => $ws ? new WritingSystemService($ws) : null));
+}
+
+export function useWritingSystemService(): WritingSystemService {
+  const writingSystemServiceContext = getContext<Readable<WritingSystemService | null>>(writingSystemContextName);
+  if (!writingSystemServiceContext) throw new Error('Writing system context is not initialized. Are you in the context of a project?');
+  const writingSystemService = get(writingSystemServiceContext);
+  if (!writingSystemService) throw new Error('Writing system service is not initialized.');
+  return writingSystemService;
+}
+
+export class WritingSystemService {
+
+  constructor(private readonly writingSystems: IWritingSystems) { }
+
+  allWritingSystems(selection: Extract<WritingSystemSelection, 'vernacular-analysis' | 'analysis-vernacular'> = 'vernacular-analysis'): IWritingSystem[] {
+    return this.pickWritingSystems(selection);
+  }
+
+  get analysis(): IWritingSystem[] {
+    return this.pickWritingSystems('analysis');
+  }
+
+  get vernacular(): IWritingSystem[] {
+    return this.pickWritingSystems('vernacular');
+  }
+
+  defaultVernacular(): IWritingSystem | undefined {
+    return this.writingSystems.vernacular[0];
+  }
+
+  defaultAnalysis(): IWritingSystem | undefined {
+    return this.writingSystems.analysis[0];
+  }
+
+  pickWritingSystems(
+    ws?: WritingSystemSelection,
+  ): IWritingSystem[] {
+    ws = ws ?? 'vernacular-analysis';
+    switch (ws) {
+      case 'vernacular-analysis':
+        return [...new Set([...this.writingSystems.vernacular, ...this.writingSystems.analysis].sort())];
+      case 'analysis-vernacular':
+        return [...new Set([...this.writingSystems.analysis, ...this.writingSystems.vernacular].sort())];
```

Comment:
I know you copied these, but I don't think this set (or sort) is doing anything, the set won't deduplicate because each of these objects came from json, so even if they have the same data they won't be the same instance. Can't this just be simplified greatly?

Reply:
Sounds good. Yeah, I don't know why those sets are there 🤷